apis/mutations/v1alpha1/match.go

```diff
@@ -0,0 +1,42 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Match struct {
+	Kinds              []Kinds                            `json:"kinds,omitempty"`
+	Scope              apiextensionsv1beta1.ResourceScope `json:"scope" protobuf:"bytes,4,opt,name=scope,casttype=ResourceScope"`
+	Namespaces         []string                           `json:"namespaces,omitempty"`
+	ExcludedNamespaces []string                           `json:"excludedNamespaces,omitempty"`
+	LabelSelector      *metav1.LabelSelector              `json:"labelSelector,omitempty"`
+	NamespaceSelector  *metav1.LabelSelector              `json:"namespaceSelector,omitempty"`
+}
+
+// Kinds accepts a list of objects with apiGroups and kinds fields
+// that list the groups/kinds of objects to which the mutation will apply.
+// If multiple groups/kinds objects are specified,
+// only one match is needed for the resource to be in scope.
+type Kinds struct {
+	// APIGroups is the API groups the resources belong to. '*' is all groups.
+	// If '*' is present, the length of the slice must be one.
+	// Required.
+	APIGroups string `json:"apiGroups,omitempty" protobuf:"bytes,1,rep,name=apiGroups"`
```

Comment:
I think both this and the next one must be []string to be aligned with Constraints.
@maxsmythe can you confirm?

Reply:
Ah, you are correct, both of these should be `[]string`

Reply:
Done